lib/Connection.js

```diff
@@ -1,217 +1,222 @@
-  }
-
-  dispose() {
-    this._onClose();
-    this._ws.close();
-  }
-
-  /**
-   * @param {string} targetId
-   * @return {!Promise<!Session>}
-   */
-  async createSession(targetId) {
-    const {sessionId} = await this.send('Target.attachToTarget', {targetId});
-    const session = new Session(this, targetId, sessionId);
-    this._sessions.set(sessionId, session);
-    return session;
-  }
-}
-
-class Session extends EventEmitter {
-  /**
-   * @param {!Connection} connection
-   * @param {string} targetId
-   * @param {string} sessionId
-   */
-  constructor(connection, targetId, sessionId) {
-    super();
-    this._lastId = 0;
-    /** @type {!Map<number, {resolve: function, reject: function, method: string}>}*/
-    this._callbacks = new Map();
-    this._connection = connection;
-    this._targetId = targetId;
-    this._sessionId = sessionId;
-  }
-
-  /**
-   * @return {string}
-   */
-  targetId() {
-    return this._targetId;
-  }
-
-  /**
-   * @param {string} method
-   * @param {!Object=} params
-   * @return {!Promise<?Object>}
-   */
-  send(method, params = {}) {
-    if (!this._connection)
-      return Promise.reject(new Error(`Protocol error (${method}): Session closed. Most likely the page has been closed.`));
-    const id = ++this._lastId;
-    const message = JSON.stringify({id, method, params});
-    debugSession('SEND ► ' + message);
-    this._connection.send('Target.sendMessageToTarget', {sessionId: this._sessionId, message}).catch(e => {
-      // The response from target might have been already dispatched.
-      if (!this._callbacks.has(id))
-        return;
-      const callback = this._callbacks.get(id);
-      this._callbacks.delete(id);
-      callback.reject(e);
-    });
-    return new Promise((resolve, reject) => {
-      this._callbacks.set(id, {resolve, reject, method});
-    });
-  }
-
-  /**
-   * @param {string} message
-   */
-  _onMessage(message) {
-    debugSession('◀ RECV ' + message);
-    const object = JSON.parse(message);
-    if (object.id && this._callbacks.has(object.id)) {
-      const callback = this._callbacks.get(object.id);
-      this._callbacks.delete(object.id);
-      if (object.error)
-        callback.reject(new Error(`Protocol error (${callback.method}): ${object.error.message} ${object.error.data}`));
-      else
-        callback.resolve(object.result);
-    } else {
-      console.assert(!object.id);
-      this.emit(object.method, object.params);
-    }
-  }
-
-  async dispose() {
-    console.assert(!!this._connection, 'Protocol error: Connection closed. Most likely the page has been closed.');
-    await this._connection.send('Target.closeTarget', {targetId: this._targetId});
-  }
-
-  _onClosed() {
-    for (const callback of this._callbacks.values())
-      callback.reject(new Error(`Protocol error (${callback.method}): Target closed.`));
-    this._callbacks.clear();
-    this._connection = null;
-  }
-}
-
-module.exports = {Connection, Session};
+/**

```

Comment:
Are these different line-endings? Can you please revert the change so that the actual changes are clear?

Reply:
Perhaps, certainly i will do.